File: jungyungee/Week 32/예산.py
def solution(d, budget):
    answer = 0
    sum = 0
    d.sort()
    
    for i in d:
        sum += i

    if sum <= budget:
        answer = len(d)
    else:
        while sum > budget:
            sum = sum - d[-1]
            d.pop()
        answer = len(d)
    return answer

File: un6unw/Week32/예산.java
import java.util.*;

class Solution {
    public int solution(int[] d, int budget) {
        Arrays.sort(d);  // 작은 금액부터 지원
        int count = 0;

        for (int cost : d) {
            if (budget >= cost) {
                budget -= cost;
                count++;
            } else {
                break;
            }
        }

        return count;
    }
}